city_metrix/layers/building_classifier.py

```diff
@@ -0,0 +1,181 @@
+
+        # ANBH is the average height of the built surfaces, USE THIS
+        # AGBH is the amount of built cubic meters per surface unit in the cell
+        # https://ghsl.jrc.ec.europa.eu/ghs_buH2023.php
+        anbh = (ee.ImageCollection("projects/wri-datalab/GHSL/GHS-BUILT-H-ANBH_R2023A")
+                .filterBounds(ee.Geometry.BBox(*bbox))
+                .select('b1')
+                .mosaic()
+                )
+        ds = xr.open_dataset(
+            ee.ImageCollection(anbh),
+            engine='ee',
+            scale=100,
+            crs=crs,
+            geometry=ee.Geometry.Rectangle(*bbox)
+        )
+        anbh_data = ds.b1.compute()
+        # get in rioxarray format
+        anbh_data = anbh_data.squeeze("time").transpose("Y", "X").rename({'X': 'x', 'Y': 'y'})
+        
+        anbh_1m = anbh_data.rio.reproject(
+            dst_crs=crs,
+            shape=esa_1m.shape,
+            resampling=Resampling.nearest,
+            nodata=0
+        )
+
+        return buildings_sample, esa_1m, ulu_lulc_1m, anbh_1m, crs
+    
+
+    # Extract values to buildings as coverage fractions
+    # Extract average of pixel values to buildings
+    # Reproject to local state plane and calculate area
+    def calc_majority_ULU_mean_ANBH_area(self, row, building_sample_1m, id_col, ulu_lulc_1m, anbh_1m):
+        mask = building_sample_1m == row[id_col]
+        masked_ulu = ulu_lulc_1m.values[mask]
+        
+        # Extract values to buildings as coverage fractions
+        # when there is no majority class, use 1-Non-residential as default
+        if masked_ulu.size == 0:
+            majority_ULU = 1
+        else:
+            unique, counts = np.unique(masked_ulu, return_counts=True)
+            sorted_indices = np.argsort(-counts)  # Sort by descending order
+            
+            # Apply your specific logic
+            if unique[sorted_indices[0]] != 3:
+                majority_ULU = unique[sorted_indices[0]]
+            elif len(sorted_indices) > 1:
+                majority_ULU = unique[sorted_indices[1]]
+            else:
+                majority_ULU = 1  # Default to 1 non-residential
+
+        # Extract average of pixel values to buildings
+        masked_anbh = anbh_1m.values[mask]
+        if masked_anbh.size == 0:
+            mean_ANBH = 0
+        else:
+            mean_ANBH = np.mean(masked_anbh)
+        
+        # Reproject to local state plane and calculate area
+        Area_m = row.geometry.area
+
+        return pd.Series([majority_ULU, mean_ANBH, Area_m])
+        
+        # TODO
+        # roof slope model
+    def rasterize_building(self, gdf, snap_to):
+        raster = make_geocube(
+            vector_data=gdf,
+            measurements=["Value"],
+            like=snap_to,
+            fill=0
+        ).Value
+
+        return raster.rio.reproject_match(snap_to)
+    
+    def building_class_tree(self):
+        buildings_sample, esa_1m, ulu_lulc_1m, anbh_1m, crs = self.get_data()
```

Comment:
could you provide some comments on what this building classifier is classifying, and what the output is?